Easydict/NewApp/View/SettingView/Tabs/View/Shortcut/AppShortcutSetting.swift

```diff
@@ -0,0 +1,118 @@
+                }
+                HStack {
+                    Text("retry")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .retry, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("toggle_languages")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .toggle, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("pin")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .pin, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("hide")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .hide, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("shortcut_increase_font")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .increaseFontSize, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("shortcut_decrease_font")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .decreaseFontSize, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("open_in_google")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .google, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("open_in_eudic")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .eudic, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+                HStack {
+                    Text("open_in_apple_dictionary")
+                    Spacer()
+                    KeyHolderWrapper(shortcutType: .appleDic, confictAlterMessage: $confictAlterMessage).frame(width: 180, height: 24)
+                }
+            } header: {
+                Text("app_shortcut_setting")
+            }
+
+            .alert(String(localized: "shortcut_confict \(confictAlterMessage.title)"),
+                   isPresented: showAlter,
+                   presenting: confictAlterMessage)
+            { _ in
+                Button(String(localized: "shortcut_confict_confirm")) {
+                    confictAlterMessage = ShortcutConfictAlertMessage(title: "", message: "")
+                }
+            } message: { message in
+                Text(message.message)
+            }
+        }
+    }
+}
```

Comment:
```suggestion
@available(macOS 13, *)
extension ShortcutTab {
    struct AppShortcutSettingView: View {
        @State var confictAlterMessage: ShortcutConfictAlertMessage = .init(title: "", message: "")

        var body: some View {
            let showAlter = Binding<Bool>(
                get: {
                    !confictAlterMessage.message.isEmpty
                }, set: { _ in
                }
            )

            Section {
                ShortcutSettingRow(title: "shortcut_clear_input", shortcutType: .clearInput, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_clear_all", shortcutType: .clearAll, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_copy", shortcutType: .copy, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_copy_first_teanslated_text", shortcutType: .copyFirstResult, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_focus", shortcutType: .focus, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_play", shortcutType: .play, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "retry", shortcutType: .retry, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "toggle_languages", shortcutType: .toggle, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "pin", shortcutType: .pin, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "hide", shortcutType: .hide, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_increase_font", shortcutType: .increaseFontSize, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "shortcut_decrease_font", shortcutType: .decreaseFontSize, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "open_in_google", shortcutType: .google, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "open_in_eudic", shortcutType: .eudic, confictAlterMessage: $confictAlterMessage)
                ShortcutSettingRow(title: "open_in_apple_dictionary", shortcutType: .appleDic, confictAlterMessage: $confictAlterMessage)
            } header: {
                Text("app_shortcut_setting")
            }
            .alert(String(localized: "shortcut_confict \(confictAlterMessage.title)"),
                   isPresented: showAlter,
                   presenting: confictAlterMessage)
            { _ in
                Button(String(localized: "shortcut_confict_confirm")) {
                    confictAlterMessage = ShortcutConfictAlertMessage(title: "", message: "")
                }
            } message: { message in
                Text(message.message)
            }
        }

        struct ShortcutSettingRow: View {
            let title: String
            let shortcutType: ShortcutType
            @Binding var confictAlterMessage: ShortcutConfictAlertMessage

            var body: some View {
                HStack {
                    Text(title)
                    Spacer()
                    KeyHolderWrapper(shortcutType: shortcutType, confictAlterMessage: $confictAlterMessage)
                        .frame(width: 180, height: 24)
                }
            }
        }
    }
}
```

There is too much repetitive code, it needs to be optimized.

Reply:
I didn't thoroughly check this code, it's an improved code provided by gpt-4.

Reply:
okay I will try encapsulation the same code